test/inductor/test_flex_attention.py

```diff
@@ -2372,6 +2367,23 @@ def forward(self, L_query_: "f64[2, 2, 128, 4]", L_key_: "f64[2, 2, 128, 4]", L_
         l_block_mask_full_q_num_blocks = L_block_mask_full_q_num_blocks
         l_block_mask_full_q_indices = L_block_mask_full_q_indices
 
+        getattr_1 = l_query_.is_nested;  getattr_1 = None
+        getattr_2 = l_key_.is_nested;  getattr_2 = None
+        getattr_3 = l_key_.is_nested;  getattr_3 = None
+        getattr_4 = l_value_.is_nested;  getattr_4 = None
+
+        getattr_5 = l_query_.is_nested;  getattr_5 = None
+
+        getattr_6 = l_key_.is_nested;  getattr_6 = None
+
+        getattr_7 = l_value_.is_nested;  getattr_7 = None
+
+        getattr_8 = l_query_.is_nested;  getattr_8 = None
+
+        getattr_9 = l_query_.is_nested;  getattr_9 = None
+
+        getattr_10 = l_query_.is_nested;  getattr_10 = None
```

Comment:
this is super annoying - can't we drop these nodes? are they left in because we don't know if `is_nested` has side effects or not?

Reply:
I wouldn't expect that we'd need guards since `is_nested` is a property of a normal tensor and nothing is being done with the return value, but correct me if I'm wrong.

Reply:
As discussed offline, I ran benchmarking before and after this PR
```
python benchmarks/transformer/score_mod.py --decoding
```

Before:
```
| Type    |   Speedup | score_mod     | mask_mod   | dtype          | shape(B,Hq,M,Hkv,N,D)    |
|---------|-----------|---------------|------------|----------------|--------------------------|
| Average |     2.743 |               |            |                |                          |
| Max     |    10.935 | relative_bias | None       | torch.bfloat16 | (16, 16, 1, 2, 4096, 64) |
| Min     |     0.256 | None          | causal     | torch.bfloat16 | (2, 16, 1, 16, 512, 128) |
```

After:
```
| Type    |   Speedup | score_mod     | mask_mod   | dtype          | shape(B,Hq,M,Hkv,N,D)    |
|---------|-----------|---------------|------------|----------------|--------------------------|
| Average |     2.710 |               |            |                |                          |
| Max     |    10.963 | relative_bias | None       | torch.bfloat16 | (16, 16, 1, 2, 4096, 64) |
| Min     |     0.270 | None          | causal     | torch.bfloat16 | (2, 16, 1, 16, 512, 128) |
```

From what I can tell, speedup number differences are within natural variance.